### PR TITLE
Fix an error in build-heap section.

### PR DESCRIPTION
**Pull request by xiaohanyu**
If the last level of a complete binary tree can contain at most $2^{p - 1}$
nodes, then the total nodes of the binary tree is

\[
\sum_{i = 0}^{p - 1} 2^{i} = 2^{p} - 1
\]

So it should be $2^{p} - 1 \leq n$ instead of $2^{p} \leq n$.

An example:

```
     7
   /   \
  6     5
 / \   / \
4   3 2   1
```

This binary has 7 nodes, the last level has 4 nodes, so $p = 3$, and $n
= 7$, it should be $2^3 - 1 \leq 7$ instead of $2^3 \leq 7$.

### CODE REVIEW

**Review by liuxinyu95 — APPROVED** [review]

**Comment by xiaohanyu** [comment]
Seems you should merge this after approve?

**Pull request merged**
